03-data_visualisation.Rmd

```diff
@@ -1,14 +1,164 @@
 
-- Jon will try to prefill these ahead of each week.
+- **Load** the `tidyverse` family of packages.
+- Produce a simple plot with **{ggplot2}.**
+- Use **`aes`thetic mappings** to produce more complex plots.
+- Deal with **common R programming problems.**
+- Produce **small multiples** with `facet()`.
+- Combine multiple `geom_*()` objects to produce more complex plots.
+- Recognize the interaction between `stat`s and `geom`s.
+- Use the `position` argument to control **data layout.**
+- Use alternative **coordinate systems** for plots in ggplot2.
+- Describe the components of the **layered grammar of graphics.**
+
+## Loading Packages in R
+
+- Use `install.packages("PACKAGE_NAME")` to **install** a package in R.
+  - You need to do this before you can use a package, but you only need to do it once.
+  - Depending where you're learning R, the packages you need may already be installed.
+- Use `library(PACKAGE_NAME)` to **load** a package in R.
+  - In general, you want to do this at the start of any session where you use a package.
+  - Alternatively, you can refer to a package every time you use it, such as `ggplot2::ggplot()`. "ggplot2" is the package, "ggplot()" is the function call, "::" tells R "look up this function in this package."
+
+```{r library}
+library(tidyverse)
+```
+
+## First Steps
+
+The `{ggplot2}` package lets us build up plots layer by layer.
+
+```{r graphing-template-1, eval = FALSE}
+ggplot(data = <DATA>) + 
+  <GEOM_FUNCTION>(mapping = aes(<MAPPINGS>))
+```
+
+- `ggplot(data = <DATA>)`: Set up a base plot with data.
+- `<GEOM_FUNCTION>(mapping = aes(<MAPPINGS>))`: Add a geometry, such as a point (`geom_point()`).
+  - `mapping`: Map visual properties (`x`, `y`) to variables from `data` (`displ`, `hwy`).
+  - `aes()`: A function to do the mapping. The name is short for "aesthetics." **Spoiler alert:** "aesthetics" is the word Hadley uses for "visual properties."
+- Other little things brought out here:
+  - Type `?FUNCTION_NAME` (`?) to load help for that function (or data object).
+  - All `{tidyverse}` help is also available at [tidyverse.org](https://www.tidyverse.org/).
+  - Super minor convention: "`hwy` vs `cyl`" means "`hwy` as `y` and `cyl` as `x`." 
+  - Solutions are available at [jrnold.github.io/r4ds-exercise-solutions](https://jrnold.github.io/r4ds-exercise-solutions/)
+
+
+## Aesthetic mappings
+
+- **Aesthetics** are visual properties of objects in the plot.
+- Includes:
+  - size
+  - shape
+  - color
+  - alpha (transparency)
+  - x
+  - y
+  - others, sometimes specific to a geom.
+- Use the `mapping` argument and the `aes` function to map an aesthetic to a variable in `data`.
+- Or assign the value outside of `mapping` to set it for everything:
+  - `geom_point(mapping = aes(x = displ, y = hwy), color = "blue")`
+  - A super common error: Trying to do that *inside* the `aes` call. Results in random effects because `aes()` automatically figures out the necessary **scale** for your data.
```

Comment:
Trying to do what? The color?

Reply:
actually, on a second look, I don't think it's that ambiguous. So never mind. :)

Reply:
The "that" is "this thing that's in this family of bullets." I *think* it's clear-enough to present.